dev/Tiles/entity.h

```diff
@@ -0,0 +1,66 @@
+#include <stdlib.h>
+#include <stdbool.h>
+#include "metatile.h"
```

Comment:
I might question why you'd want to split your API into 3 different .h files: it only really makes sense if you expect your consumers to regularly use parts of the lib and not others. Otherwise it's just more jumping around to reference code and / or more include lines to add.

See the way SMSlib is structured for an example: there's a single .h file and many .c files:
https://github.com/sverx/devkitSMS/tree/master/SMSlib/src

As we discussed in Discord, it's strange to put implementation into .h files, but you said it's just for rapidity of development, which I get. Fully formed .h files for libs often need some boilerplate code in to ensure they are not being included twice, especially if they include implementation.

Reply:
Point noted.👍